File: models/claims/nhisClaimExport.js



// models/claims/nhisClaimExport.js
const { DataTypes } = require('sequelize');
const sequelize = require('../../config/database');
const Institution = require('../institution');
const Staff = require('../staff');
// const User = require('../users/users'); // adjust to your actual user model
// const Institution = require('../institution');

const NHISClaimExport = sequelize.define('NHISClaimExport', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  batch_number: {
    type: DataTypes.STRING,
    allowNull: false,
    unique: true,
  },
  institution_id: {
    type: DataTypes.UUID,
    allowNull: false,
    references: {
      model: Institution,
      key: 'id'
    }
  },
  file_name: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  file_path: {
    type: DataTypes.STRING,
    allowNull: true,
  },
  total_claims: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
  },
  total_amount: {
    type: DataTypes.DECIMAL(15, 2),
    defaultValue: 0.00,
  },
  generated_by: {
    type: DataTypes.UUID,
    allowNull: true,
    references: {
      model: Staff,
      key: 'id'
    }
  },
  export_status: {
    type: DataTypes.ENUM('Pending', 'Submitted', 'Approved', 'Rejected'),
    defaultValue: 'Pending'
  },
  remarks: {
    type: DataTypes.TEXT,
    allowNull: true,
  }
}, {
  tableName: 'nhis_claim_exports',
  timestamps: true,
});

// associations
NHISClaimExport.associate = (models) => {
  NHISClaimExport.belongsTo(models.Institution, { foreignKey: 'institution_id', as: 'institution' });
  NHISClaimExport.belongsTo(models.Staff, { foreignKey: 'generated_by', as: 'generator' });
  NHISClaimExport.hasMany(models.NHISClaimExportClaim, { foreignKey: 'export_id', as: 'exported_claims' });
}

module.exports = NHISClaimExport;

File: claims-service/app/models/nhis_claim_export.py
"""NHISClaimExport model - tracks NHIA XML export batches."""

import uuid
from datetime import datetime

from sqlalchemy import Column, DateTime, Enum, Float, Integer, String, Text
from sqlalchemy.dialects.postgresql import UUID

from app.database import Base
from app.utils.types import ExportStatus


class NHISClaimExport(Base):
    __tablename__ = "nhis_claim_exports"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    batch_number = Column(String(100), unique=True, nullable=False)
    institution_id = Column(UUID(as_uuid=True), nullable=False, index=True)
    file_name = Column(String(255), nullable=False)
    file_path = Column(String(500), nullable=True)
    total_claims = Column(Integer, default=0)
    total_amount = Column(Float, default=0)
    generated_by = Column(UUID(as_uuid=True), nullable=True)
    export_status = Column(
        Enum(ExportStatus, name="export_status_enum", create_constraint=False),
        default=ExportStatus.PENDING,
    )
    remarks = Column(Text, nullable=True)

    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
    updated_at = Column(DateTime(timezone=True), default=datetime.utcnow, onupdate=datetime.utcnow)
